build.gradle

```diff
@@ -83,16 +86,137 @@ subprojects {
     }
 }
 
+dockerCompose {
+    environment.put "EVENTUATE_COMMON_VERSION", eventuateCommonImageVersion
+    environment.put "EVENTUATE_ACTIVE_MQ_VERSION", eventuateActiveMQImageVersion
+    environment.put "EVENTUATE_RABBIT_MQ_VERSION", eventuateRabbitMQImageVersion
+
+    removeOrphans = true
+
+    mysql {
+        projectName = null
+        useComposeFiles = ["docker-compose-mysql.yml"]
+        removeContainers = project.ext.removeContainers
+    }
+
+    mysqlmigration {
+        projectName = null
+        removeContainers = project.ext.removeContainers
+        useComposeFiles = ["docker-compose-eventuate-local-mysql-for-migration.yml"]
+        startedServices = System.env.COMPOSE_SERVICES == null || System.env.COMPOSE_SERVICES == '' ? [] :"$System.env.COMPOSE_SERVICES".split(",")
+    }
+
+    mariadb {
+        projectName = null
+        useComposeFiles = ["docker-compose-mariadb.yml"]
+        removeContainers = project.ext.removeContainers
+    }
+
+    mssql {
+        projectName = null
+        useComposeFiles = ["docker-compose-mssql.yml"]
+        removeContainers = project.ext.removeContainers
+    }
+
+    postgres {
+        projectName = null
+        useComposeFiles = ["docker-compose-postgres.yml"]
+        removeContainers = project.ext.removeContainers
+    }
+
+    mysqljson {
+        projectName = null
+        useComposeFiles = ["docker-compose-mysql-json.yml"]
+        removeContainers = project.ext.removeContainers
+    }
+
+    mariadbjson {
+        projectName = null
+        useComposeFiles = ["docker-compose-mariadb-json.yml"]
+        removeContainers = project.ext.removeContainers
+    }
+
+    mssqljson {
+        projectName = null
+        useComposeFiles = ["docker-compose-mssql-json.yml"]
+        removeContainers = project.ext.removeContainers
+    }
+
+    postgresjson {
+        projectName = null
+        useComposeFiles = ["docker-compose-postgres-json.yml"]
+        removeContainers = project.ext.removeContainers
+    }
+
+    mysqlbinlog {
+        projectName = null
+        removeContainers = project.ext.removeContainers
+        useComposeFiles = ["docker-compose-mysql.yml", "docker-compose-cdc-mysql-binlog.yml"]
+        startedServices = System.env.COMPOSE_SERVICES == null || System.env.COMPOSE_SERVICES == '' ? [] :"$System.env.COMPOSE_SERVICES".split(",")
```

Comment:
Rather than  repeatedly copy/paste of `System.env.COMPOSE_SERVICES == null || System.env.COMPOSE_SERVICES == '' ? [] :"$System.env.COMPOSE_SERVICES".split(",")`

You can define a variable:

```
diff --git build.gradle build.gradle
index 0dc4da4..f35054e 100644
--- build.gradle
+++ build.gradle
@@ -55,10 +55,11 @@ task compileAll(type: GradleBuild) {
   tasks = ["testClasses", "compileIntegrationTestJava", "compileComponentTestJava"]
 }
 
+def startedServicesAsList = project.hasProperty('startedServices') ?
+        project.ext.startedServices.split(',') : []
+
 dockerCompose {
   projectName = null
   removeOrphans = true
-  if (project.hasProperty('startedServices'))
-    startedServices= project.ext.startedServices.split(',')
-
+  startedServices = startedServicesAsList
 }
```

Reply:
ok